.github/workflows/build.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest pytest-cov pycodestyle
+        python -m pip install pip install .
```

Comment:
I noticed that this workflow is called "**Build**", but actually, there is no section `build` here, rather the package is installed in the section "`Install dependencies`". I suggest creating a new section `-name: Build` and build the package using `python -m build` command. To do that, `build` should be installed in `Install dependencies` as `pip install build`. 
Also, I saw that `flake8 pytest pytest-cov pycodestyle` are included here as dependencies for the `Build`.  But actually these packages are not required for build but rather for testing or development. I suggest removing them form here. Therefore, the workflow can be written as:
```yaml

    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        python -m pip install build
    - name: Build
      run: |
        python -m build
```